code/modules/wiremod/components/math/arithmetic.dm

```diff
@@ -68,9 +48,17 @@
 					result = null
 					break
 				result /= value
+			if(COMP_ARITHMETIC_MODULO)
+				//Another protect from divide by zero.
+				if(value == 0)
+					result = null
+					break
+				//BYOND's built in modulus operator doesn't work well with decimals, so I'm using this method instead
+				var/multiples = round(result / value)
+				result -= multiples * value
 			if(COMP_ARITHMETIC_MAX)
 				result = max(result, value)
 			if(COMP_ARITHMETIC_MIN)
 				result = min(result, value)
 
-	output.set_output(result)
+	return result
```

Comment:
you can replace `result` with `.` in this proc

Reply:
That makes, sense, I'll do that